Extension/src/LanguageServer/ui_new.ts

```diff
@@ -403,6 +416,24 @@ export class NewUI implements UI {
         }
     }
 
+    private compilerTimout?: NodeJS.Timeout;
+    public async showCompilerStatusIcon(show: boolean): Promise<void> {
+        if (!telemetry.showStatusBarIntellisenseIndicator()) {
+            return;
+        }
+        if (this.compilerTimout) {
+            clearTimeout(this.compilerTimout);
+        }
+        if (show) {
+            this.compilerTimout = setTimeout(() => {
+                this.compilerStatusItem.show();
```

Comment:
Doesn't it need to re-check if the condition still exists to show it, since the user could have already set it in before the 15 seconds?